plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -213,6 +215,10 @@ public class ClickHouseClient
 
         Domain simplifiedDomain = domain.simplify(getDomainCompactionThreshold(session));
         if (!simplifiedDomain.getValues().isDiscreteSet()) {
+            // Push down inequality predicate if domain was not simplified
```

Comment:
Why not just use `io.trino.plugin.jdbc.PredicatePushdownController#FULL_PUSHDOWN` directly and remove `CLICKHOUSE_PUSHDOWN_CONTROLLER` ?

Reply:
because of logic within this `if`:
```
            // Push down inequality predicate if domain was not simplified
            if (domain == simplifiedDomain) {
                return FULL_PUSHDOWN.apply(session, domain);
            }
            // Domain#simplify can turn a discrete set into a range predicate
            return DISABLE_PUSHDOWN.apply(session, domain);
```
If inequality was simplified to range - this leads to incorrect results.

Reply:
You've mentioned in description that ClickHouse is case-sensitive with exactly the same byte collation as Trino and that "we can pushdown all operations on varchars".
So I don't get why this scenario requires special handling, it sounds like FULL_PUSHDOWN should always work.
I also don't understand why there is a distinction between a range predicate produced due to a domain simplification vs a non-discrete values predicate from the input. I think both should be handled the same way (i.e. full pushdown or no pushdown for both), e.g. see `io.trino.plugin.jdbc.PredicatePushdownController#CASE_INSENSITIVE_CHARACTER_PUSHDOWN`

Reply:
https://github.com/trinodb/trino/pull/23516